pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -155,15 +162,19 @@ def __init__(
         save_top_k: Optional[int] = None,
         save_weights_only: bool = False,
         mode: str = "min",
-        period: int = 1,
+        every_n_epochs: int = 1,
+        every_n_batches: int = 0,
+        period: Optional[int] = None,
     ):
         super().__init__()
         self.monitor = monitor
         self.verbose = verbose
         self.save_last = save_last
         self.save_top_k = save_top_k
         self.save_weights_only = save_weights_only
-        self.period = period
+        self.every_n_epochs = period or every_n_epochs
+        self.period = self.every_n_epochs
+        self.every_n_batches = every_n_batches
```

Comment:
will this be train or validation batches? (I assume train, just for clarification)

Reply:
train batches, since we assume the model/trainer state is the same during validation